app/src/main/java/com/nevilleantony/prototype/downloadmanager/DownloadService.java

```diff
@@ -12,49 +12,61 @@
 import androidx.annotation.Nullable;
 import androidx.core.app.NotificationCompat;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
 import io.reactivex.rxjava3.disposables.Disposable;
 import io.reactivex.rxjava3.subjects.PublishSubject;
 
 public class DownloadService extends Service {
-    private static final int NOTIFICATION_ID = 1;
     private static final int NOTIFICATION_TIMEOUT = 500;
     private static final String TAG = "Download Service";
+    private static int NOTIFICATION_ID = -1;
+    private static Map<String, NotificationCompat.Builder> notificationBuilder = new HashMap<String, NotificationCompat.Builder>();
+    private static NotificationManager notificationManager = null;
+    private static NotificationChannel channel = null;
+    private static Map<String, Integer> notificationIdMap = new HashMap<String, Integer>();
     private PendingIntent actionIntent = null;
-    private NotificationCompat.Builder notificationBuilder = null;
-    private NotificationManager notificationManager = null;
     private Notification notification = new Notification();
```

Comment:
This field can be removed and instead wherever its used we can use a local variable